src/client/rest_client/loginClient.js

```diff
@@ -54,24 +42,16 @@ export default class LoginClient {
      * @return {Promise} - resolves if successfully added user, rejects otherwise.
      */
     register(userId, password, email) {
-        let url = this.baseUrl + '/api/register',
-            data = {
-                userId: userId,
-                password: password,
-                email: email || ''
-            };
+        let data = {
+            userId: userId,
+            password: password,
+            email: email || ''
+        };
 
-        return new Promise((resolve, reject) => {
-            superagent
-                .post(url)
-                .send(data)
-                .end((err, res) => {
-                    if (err || !res.ok) {
-                        reject(err);
-                    } else {
-                        resolve(res.body);
-                    }
-                });
-        });
+        return super.post(['/api', 'register'], data);
+    }
+
+    getAllowGuests() {
```

Comment:
This could be called getGmeConfig and be reused when checking 'allowUserRegistration' too.